src/python/pants/backend/helm/goals/lint.py

```diff
@@ -42,7 +43,9 @@ async def partition_helm_lint(
         field_set for field_set in request.field_sets if not field_set.skip_lint.value
     )
     charts = await MultiGet(Get(HelmChart, HelmChartRequest(field_set)) for field_set in field_sets)
-    return Partitions((chart, (field_set,)) for chart, field_set in zip(charts, field_sets))
+    return Partitions(
+        Partition(chart, (field_set,)) for chart, field_set in zip(charts, field_sets)
+    )
```

Comment:
We could make the constructor of `Partitions` also accept tuples and create the `Partition` object, much like a `dict` constructor.

Reply:
I poked at this a little bit, but my preference would be to leave it as-is. The default rules & `Partitions.single_partition` should help most use-cases avoid the boilerplate. In cases where it can't be avoided the extra characters will (hopefully) be small compared to the logic of the custom partitioner - so it'll be not-so-annoying and maximally clear what's going on.